lib/patientFinder/utils.rb

```diff
@@ -0,0 +1,165 @@
+        AMBIGUOUS_PATIENTS[["fed089904c10b81c036adddedddebe7b", "846c1c2ba8370c2f5504b315cc4b1d5d", "b5633133e3421216ced2bdef4dbf382d", "0a26a978d07240b0f917e96727db31d8","ee85c89f24946e2ddca12c6edc5181dc","b54a4e3ab37de7e5f8094793afb8a699","2678a4e396aaec03b860d5aeadcad8e6","929dd1f2c4c54c024fd4d18b1307fdb1","fdea0c22270417d9e59f20b07f642679","91bd37f9cebf7b6ef9f72d7fd6148a81","0085074bb549ffefffa6e16ff34df140","343731b99c6bdb35a1fc31ce9ed6f889","eb6af018e11320f5c163624beb83767d","470f57b022eaeffd4d599078e851a56d","0dbaf9336f7aa1590265250a0eebe548","29cecb1da08efd331fce823e12b607d5","d156a6d38e10efc30eda3cace7456537","8130b2ff5774f1593c86eba8dca4c37b","697e147f076648275e518e7b3ff41dcd"]] = "fed089904c10b81c036adddedddebe7b"       
+        
+        #find matches for test patients
+        def self.get_MRNs(master_records, test_patients)
+    
+            #store the results
+            tests_to_MRNs= Hash.new
+            
+            #iterate through master records and find candidates 
+            test_patients.each { |test_patient|
+                #each test patient might have one or more MRN candidates
+                candidates = get_candidates_for_test(master_records, test_patient.record)
+                
+                #resolve cadidates to unique MRNs - with other words resolve the candidates
+                get_MRN_for_tests(candidates.map(&:medical_record_number), test_patient, tests_to_MRNs)
+            }
+    
+            tests_to_MRNs
+        end
+        
+        #resolves ambiguos candidates to a unique MRN 
+        def self.get_MRN_for_ambiguous(candidate_MRNs)
+            AMBIGUOUS_PATIENTS.each_pair {  |ambiguous_entries, mrn| 
+                return mrn if (candidate_MRNs-ambiguous_entries).empty?
+            }
+            "too ambiguous #{candidate_MRNs}"
+        end
+    
+        #resolves the candidates to the test files (one test file - one candidate - when multiple candidates are found manual anotation is used)
+        def self.get_MRN_for_tests(candidate_MRNs, test_record, test_to_MRN_hash)
+            
+            if(candidate_MRNs.length==1)
+                test_to_MRN_hash[test_record.fileName]=candidate_MRNs.first
+            else
+                test_to_MRN_hash[test_record.fileName]=get_MRN_for_ambiguous(candidate_MRNs)
+            end
+        end    
+    
+        #finds sections with data for a record (out of a array of given sections)    
+        # def self.get_non_empty_sections_for_record(sections, record)
+        #     sections_with_data=Array.new
+        #     sections.each{ |section|
+        #         if !(record[section].nil? || record[section].empty?)
+        #             sections_with_data<< section
+        #         end
+        #     }
+        #     sections_with_data
+        # end
```

Comment:
This method can be implemented this way:

    def self.get_non_empty_sections_for_record(sections, record)
        sections.select { |section| record[section].present? }
    end

That being the case, I would probably just inline it where it is being called in `get_candidates_for_test`.

This implementation is not completely identical, because `.present?` is the inverse of `.blank?`, which is [slightly different](https://stackoverflow.com/questions/885414/a-concise-explanation-of-nil-v-empty-v-blank-in-ruby-on-rails) from `.nil? || .empty?`, but those differences should not be relevant in this case.